manifests/charts/istio-control/istio-discovery/templates/service.yaml

```
@@ -3,6 +3,10 @@ kind: Service
 metadata:
   name: istiod{{- if not (eq .Values.revision "") }}-{{ .Values.revision }}{{- end }}
   namespace: {{ .Release.Namespace }}
+  annotations:
+  {{- if .Values.pilot.serviceAnnotations }}
+{{ toYaml .Values.pilot.serviceAnnotations | indent 4 }}
+  {{- end }}
```

This avoids creating empty annotation [like](https://github.com/istio/istio/pull/35202/files#diff-5d70e6b65d86515abc1bcdf047a0cbbadff8858086ba2ea07ad85b4c8c274bdeR1416).
```suggestion
  {{- if .Values.pilot.serviceAnnotations }}
  annotations:
{{ toYaml .Values.pilot.serviceAnnotations | indent 4 }}
  {{- end }}
```